doc/reference/peripherals/rtio.rst

```diff
@@ -0,0 +1,48 @@
+.. _rtio_interface:
+
+Real-Time IO
+############
+
+The real-time I/O or RTIO subsystem exposes an API for efficiently
+moving information in a latency and processing friendly way, between
+a Zephyr application and external I2C or SPI devices (such as sensors).
+
+Overview
+********
+
+RTIO provides two significant C structures and APIs for moving bytes around
+efficiently, pipes and blocks.
+
+Pipes, defined by :c:type:`struct rtio_pipe`, are used to move blocks in a
+pollable manner from a source to a sink of data. A pipe is effectively a
+single-input single-output queue of blocks.
+
+Blocks, defined by :c:type:`struct rtio_block`, provide the unit of data
+being transferred through a pipe. These blocks contain a byte layout
+identifier provided by the source of the block, timestamps for when the block
+began and finished being created, as well as length and size in bytes.
+
+Blocks are used to allow for simple DMA transfers as well as controlling the
+latency between a source and sink.
+
+Sources can be sensors such has a high rate accelerometer, gyro, or
+magnetometer. In these cases the driver will typically provide blocks
+encoded in the sensors native format which is often times the most compact
+representation.
+
+Intermediary processing steps representing both sinks and sources of data are
+also possible. One common task is taking the raw byte layout coming from a
+device such as a 9-DoF intertial motion sensors and converting it to SI units
+with common numerical formats for that sensor kind. Several processing steps
+might be connected together. Taking the above example of getting a common
+9-DoF SI data stream could be used to determine an earth orientation vector
+using sensor fusion.
+
+The RTIO subsystem lets you both efficiently move data around and
```

Comment:
Thanks for this documentation. It would be nice to add a few sentences about triggers to this.

Reply:
Definitely, I have some function comments on rtio_trigger but nothing in the general docs explaining what its there for.